db/db_bloom_filter_test.cc

```diff
@@ -667,6 +670,401 @@ TEST_F(DBBloomFilterTest, BloomFilterReverseCompatibility) {
+      // behavior and we don't need a large number of keys to verify we
+      // indeed charge the final filter for cache reservation, even though final
+      // filter is a lot smaller than hash entries.
+      num_key_ = 1 * CacheReservationManager::GetDummyEntrySize() /
+                 sizeof(FilterConstructionReserveMemoryHash);
+    } else {
+      // For Ribbon Filter + FullFilter case, we need a large enough number of
+      // keys so that charging final filter after releasing the hash entries
+      // reservation will trigger at least another dummy entry (or equivalently
+      // to saying, causing another peak in cache reservation) as banding
+      // reservation might not be a multiple of dummy entry.
+      num_key_ = 12 * CacheReservationManager::GetDummyEntrySize() /
+                 sizeof(FilterConstructionReserveMemoryHash);
+    }
+  }
+
+  BlockBasedTableOptions GetBlockBasedTableOptions() {
+    BlockBasedTableOptions table_options;
+
+    // We set cache capacity big enough to prevent cache full for convenience in
+    // calculation.
+    constexpr std::size_t kCacheCapacity = 100 * 1024 * 1024;
+
+    table_options.reserve_table_builder_memory = reserve_table_builder_memory_;
+    table_options.filter_policy.reset(new BloomFilterPolicy(10, policy_));
+    table_options.partition_filters = partition_filters_;
+    if (table_options.partition_filters) {
+      table_options.index_type =
+          BlockBasedTableOptions::IndexType::kTwoLevelIndexSearch;
+      // We set table_options.metadata_block_size big enough so that each
+      // partition trigger at least 1 dummy entry insertion each for hash
+      // entries and final filter.
+      table_options.metadata_block_size = 409000;
+    }
+
+    LRUCacheOptions lo;
+    lo.capacity = kCacheCapacity;
+    lo.num_shard_bits = 0;  // 2^0 shard
+    lo.strict_capacity_limit = true;
+    cache_ = std::make_shared<FilterConstructResPeakTrackingCache>(
+        (NewLRUCache(lo)));
+    table_options.block_cache = cache_;
+
+    return table_options;
+  }
+
+  std::size_t GetNumKey() { return num_key_; }
+
+  bool ReserveTableBuilderMemory() { return reserve_table_builder_memory_; }
+
+  BloomFilterPolicy::Mode GetFilterPolicy() { return policy_; }
+
+  bool PartitionFilters() { return partition_filters_; }
+
+  std::shared_ptr<FilterConstructResPeakTrackingCache>
+  GetFilterConstructResPeakTrackingCache() {
+    return cache_;
+  }
+
+ private:
+  std::size_t num_key_;
+  bool reserve_table_builder_memory_;
+  BloomFilterPolicy::Mode policy_;
+  bool partition_filters_;
+  std::shared_ptr<FilterConstructResPeakTrackingCache> cache_;
+};
+
+std::vector<bool> reserve_builder_memory_options{true, false};
+std::vector<BloomFilterPolicy::Mode> policies{
+    BloomFilterPolicy::Mode::kFastLocalBloom,
+    BloomFilterPolicy::Mode::kStandard128Ribbon,
+    BloomFilterPolicy::Mode::kDeprecatedBlock};
+std::vector<bool> partition_filters_options{false, true};
```

Comment:
In instantiating test cases, I advocate an approach like how MC/DC code coverage[*] tackles the exponential blow-up problem: for each parameter, there should be two instantiations differing in only that parameter. Obviously you want to cover the range of values, but there doesn't have to be blow-up e.g. in mostly useless reserve_builder_memory_options=false instantiations.

I suggest
{false, kFastLocalBloom, false}
{true, kFastLocalBloom, false}
{true, kFastLocalBloom, true}
{true, kStandard128Ribbon, false}
{true, kStandard128Ribbon, true}
{true, kDeprecatedBlock, false}
{true, kLegacyBloom, false}

(Yes, before I was critical of listing a bunch of instantiations, but if you trim them down strategically, I think it's better to save the testing runtime. We have some terribly slow, IMHO poorly planned tests in our suite with hundreds of instantiations.)

[*] https://en.wikipedia.org/wiki/Modified_condition/decision_coverage